_posts/2016-04-06-Testing-Schemes.markdown

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: "Using Xcode's Schemes to run a subset of your tests"
+date: 2016-04-06 12:09
+author: orta
+categories: [mobile, xcode]
+---
+
+[Eigen](https://github.com/artsy/eigen) has hit the point where testing is a chore, I [have ideas](https://github.com/orta/life/issues/71) on how to improve this, based on [Code Injection](http://artsy.github.io/blog/2016/03/05/iOS-Code-Injection/), but they aren't fully fleshed out yet and are time-intensive.
+
+On my fastest computer, we're just under a minute - `Executed 1105 tests, with 1 failure (0 unexpected) in 43.221 (48.201) seconds`  for the whole suite. I think I could probably live with 20 seconds max. So I studied how [AppCode](https://www.jetbrains.com/objc/) handles running tests, and this will be an illustrated guide as to how you can easily run the subset of tests in Xcode based on their techniques.
+
+<!-- more -->
+
+### What are Schemes?
+
+> An Xcode scheme defines a collection of targets to build, a configuration to use when building, and a collection of tests to execute.
+
+> You can have as many schemes as you want, but only one can be active at a time. You can specify whether a scheme should be stored in a project—in which case it’s available in every workspace that includes that project, or in the workspace—in which case it’s available only in that workspace. When you select an active scheme, you also select a run destination (that is, the architecture of the hardware for which the products are built).
+
+To [quote Apple](https://developer.apple.com/library/ios/featuredarticles/XcodeConcepts/Concept-Schemes.html).
+
+### Hatching a Scheme
+
+The Eigen Test Suite is around 50 test classes, these look a bit like this:
+
+![Tests](/images/2016-04-06-Testing-Schemes/tests.png)
+
+We want to be able to say, I want to only run tests for the Fairs section of Eigen, as that is where I will be working for the next few days. To get started, I need to create a new Scheme. You will have seen the schemes before when you click on the Target side of this Targer/Sim button in the top left of Xcode.
```

Comment:
Maybe put `I want to only run tests for the Fairs section of Eigen` in "quotes"?